supertokens_python/recipe/session/access_token.py

```diff
@@ -45,45 +45,29 @@ def sanitize_number(n: Any) -> Union[Union[int, float], None]:
 
 def get_info_from_access_token(
     jwt_info: ParsedJWTInfo,
-    jwk_clients: List[JWKClient],
+    combined_jwks: List[PyJWK],
     do_anti_csrf_check: bool,
 ):
     try:
         payload: Optional[Dict[str, Any]] = None
-        client: Optional[JWKClient] = None
-        keys: Optional[List[PyJWK]] = None
-
-        # Get the keys from the first available client
-        for c in jwk_clients:
-            try:
-                client = c
-                keys = c.get_latest_keys()
-                break
-            except JWKSRequestError:
-                continue
-
-        if keys is None or client is None:
-            raise PyJWKClientError("No key found")
-
-        if jwt_info.version < 3:
+        if jwt_info.version >= 3:
+            matching_key = [k for k in combined_jwks if k.key_id == jwt_info.kid][0]  # type: ignore
+            payload = jwt.decode(  # type: ignore
```

Comment:
What part handles the kid missing and refreshing case?

Reply:
Now passing the JWKClient instead of list of PyJWK which has the refresh on unknown kid logic.